error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/util/MoreJUnitMatchers.java

```diff
@@ -55,25 +55,54 @@ private MoreJUnitMatchers() {}
    * Returns the names of the JUnit value factory methods specified by the given {@link
    * org.junit.jupiter.params.provider.MethodSource} annotation.
    *
+   * <p>This method differs from {@link #getMethodSourceFactoryDescriptors(AnnotationTree,
+   * MethodTree)} in that it drops any parenthesized method parameter type enumerations. That is,
+   * method descriptors such as {@code factoryMethod()} and {@code factoryMethod(java.lang.String)}
+   * are both simplified to just {@code factoryMethod}.
+   *
    * @param methodSourceAnnotation The annotation from which to extract value factory method names.
+   * @param method The method on which the annotation is located.
    * @return One or more value factory names.
+   * @see #getMethodSourceFactoryDescriptors(AnnotationTree, MethodTree)
+   */
+  // XXX: Test this method.
```

Comment:
I realized after committing that by using this method we:
1. May attempt to replace `foo()` even though the test method references `foo(someParam)`. (Due to loss of details.)
1. May attempt to replace `foo()` even though the test method references both `foo()` and `foo(someParam)`. (Due to loss of details i.c.w. set deduplication.)